test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs

```diff
@@ -723,7 +723,8 @@ public void NullValueInCanonicalFunction()
             var result = ParseFilter("day(null)", HardCodedTestModel.TestModel, HardCodedTestModel.GetPaintingType());
 
             var typeReference = result.Expression.ShouldBeSingleValueFunctionCallQueryNode("day")
-                .Parameters.Single().ShouldBeConstantQueryNode<object>(null).TypeReference;
+                .Parameters.Single().ShouldBeConvertQueryNode(EdmPrimitiveTypeKind.DateTimeOffset)

```

Comment:
Not sure I understand how the change to `FunctionCallBinder` class results into the change to this test being required

Reply:
I guess it's because `day()` is a function and the test was checking the parameter nodes type of that function. As the arguments are now "promoted", there is an extra hop via a convert node in the middle as `day` supposedly requires a data or datetime and null is considerated as object. At least, that's my understanding but I'm far from knowning the OData inner working, it's just an extrapolation based on the very little I saw.